ngraph/frontend/onnx_import/src/op/org.openvinotoolkit/group_norm.cpp

```diff
@@ -0,0 +1,148 @@
+
+                    auto data = inputs[0];
+                    auto scale = inputs[1];
+                    auto bias = inputs[2];
+
+                    size_t num_groups =
+                        static_cast<size_t>(node.get_attribute_value<int64_t>("num_groups"));
+                    float eps = node.get_attribute_value<float>("eps", 1e-5);
+
+                    auto data_pshape = data.get_partial_shape();
+                    std::shared_ptr<ngraph::Node> data_shape_node;
+                    if (data_pshape.is_static())
+                    {
+                        auto shape = data_pshape.to_shape();
+                        data_shape_node = default_opset::Constant::create(
+                            element::u64, Shape{shape.size()}, shape);
+                    }
+                    else
+                    {
+                        data_shape_node = std::make_shared<default_opset::ShapeOf>(data);
+                    }
+                    auto data_reshaped = std::make_shared<default_opset::Reshape>(
+                        data, detail::create_group_norm_shape(data, num_groups), true);
+                    const auto reduction_axes =
+                        common::get_monotonic_range_along_node_rank(data_reshaped, 2);
+                    auto mean = std::make_shared<default_opset::ReduceMean>(
+                        data_reshaped, reduction_axes, true);
+                    auto diff = std::make_shared<default_opset::Subtract>(data_reshaped, mean);
+                    auto variance = std::make_shared<default_opset::ReduceMean>(
+                        std::make_shared<default_opset::Power>(
+                            diff, default_opset::Constant::create(element::f32, Shape{}, {2})),
+                        reduction_axes,
+                        true);
+
+                    const std::shared_ptr<ngraph::Node> eps_node =
+                        std::make_shared<default_opset::Constant>(element::f32, Shape{}, eps);
+                    const auto sqrt = std::make_shared<default_opset::Sqrt>(
+                        std::make_shared<default_opset::Add>(variance, eps_node));
+
+                    const auto& rank = data.get_partial_shape().rank();
+                    NGRAPH_CHECK(rank.is_static());
+                    auto data_rank_size = rank.get_length();
+
+                    std::shared_ptr<ngraph::Node> result =
+                        std::make_shared<default_opset::Divide>(diff, sqrt);
+                    result =
+                        std::make_shared<default_opset::Reshape>(result, data_shape_node, true);
+                    result = std::make_shared<default_opset::Multiply>(
+                        reshape::reshape_channel_shaped_node_to_nchw(scale, data_rank_size),
+                        result);
+                    result = std::make_shared<default_opset::Add>(
+                        result, reshape::reshape_channel_shaped_node_to_nchw(bias, data_rank_size));
```

Comment:
Why is the resulting sub-graph generated different from what MO does in this transformation: https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/extensions/middle/GroupNorm.py. Does MO extra conversion of MVN later on in the transformation pipeline? @mateusztabaka, have you checked that?

Reply:
It's different, because the one from MO is not correct. If you take a look at https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/normalization/MVN_1.md and the paper https://arxiv.org/pdf/1803.08494.pdf, we can't really use MVN here. There is a difference in denominator: MVN uses sqrt(x) + eps, while GroupNorm requires sqrt(x + eps). 
MO implementation works on CPU only by accident - there is a bug in MKLDNN plugin: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp#L815.
I could fix that bug, but there are few transformations in MO that rely on it right now.

Reply:
@mateusztabaka by "fixing the bug" do you mean fixing mkldnn bug or change in GroupNorm conversion? In this case we should also change InstanceNormalization from ONNX, because it should suffer from the same bug. Can we make sure that this is really critical difference in real-life models?